services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/AutoRerouteSpec.groovy

```diff
@@ -116,8 +124,57 @@ class AutoRerouteSpec extends BaseSpecification {
         flowPath != reroutedFlowPath
 
         cleanup: "Bring port involved in original path up and delete flow"
-        flow?.id && northboundService.deleteFlow(flow.id)
         flowPath?.first() && northboundService.portUp(flowPath.first().switchId, flowPath.first().portNo)
-        Wrappers.wait(5) { northboundService.getAllLinks().every { it.state != IslChangeType.FAILED } }
+        flow?.id && northboundService.deleteFlow(flow.id)
+        Wrappers.wait(discoveryInterval + 3) {
+            northboundService.getAllLinks().every { it.state != IslChangeType.FAILED }
+        }
+    }
+
+    def "Flow in 'Up' status doesn't try to reroute even though more preferable path is available"() {
+        given: "Two active switches and a flow with one alternate path at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        }
+        assumeTrue("No suiting switches found", srcSwitch && dstSwitch)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 1000
+        northboundService.addFlow(flow)
+        def flowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Make current flow path less preferable than others"
+        possibleFlowPaths.findAll { it != flowPath }.each { pathHelper.makePathMorePreferable(it, flowPath) }
+
+        when: "One of the links not used by flow goes down"
+        def involvedIsls = pathHelper.getInvolvedIsls(flowPath)
+        def islToFail = topology.islsForActiveSwitches.find { !involvedIsls.contains(it) }
+        northboundService.portDown(islToFail.srcSwitch.dpId, islToFail.srcPort)
+
+        then: "Link status becomes 'FAILED'"
+        Wrappers.wait(discoveryInterval + 3) {
+            islUtils.getIslInfo(islToFail).get().state == IslChangeType.FAILED
+        }
+
+        when: "Failed link goes up"
+        northboundService.portUp(islToFail.srcSwitch.dpId, islToFail.srcPort)
+
+        then: "Link status becomes 'DISCOVERED'"
+        Wrappers.wait(discoveryInterval + 3) {
+            islUtils.getIslInfo(islToFail).get().state == IslChangeType.DISCOVERED
+        }
+
+        and: "Flow is not rerouted and doesn't use more preferable path"
+        TimeUnit.SECONDS.sleep(rerouteDelay + 2)
```

Comment:
I think it is better to sleep for rerouteDelay -1 or -2 and then use wait for couple of seconds